daemon/inertiad/auth/permissions.go

```diff
@@ -13,8 +14,12 @@ import (
 	"github.com/ubclaunchpad/inertia/daemon/inertiad/util"
 )
 
+// ctxKey represents keys used in request contexts
+type ctxKey int
+
 const (
-	errMalformedHeaderMsg = "malformed authorization error"
+	errMalformedHeaderMsg        = "malformed authorization error"
+	ctxUsername           ctxKey = iota
```

Comment:
never worked with `iota` before, do you mind explaining how this is used?

Reply:
No problem. `iota` just assigns consecutive integers to constants (docs [here](https://github.com/golang/go/wiki/Iota)). The reason I'm doing this is because Go's `Context` requires that keys with which values are associated are their own special type (they can't just be strings like in a regular `map`). This assignment is basically just setting `ctxUsername = 0`.

Reply:
ahh i see, thanks!